tests/Unit/NumericalAlgorithms/LinearOperators/Test_WeakDivergence.cpp

```diff
@@ -0,0 +1,365 @@
+      }
+    }
+  });
+}
+
+template <size_t Dim>
+void test_weak_divergence_constant_jacobian(const Mesh<Dim>& mesh) {
+  CAPTURE(Dim);
+  CAPTURE(mesh);
+  CAPTURE(mesh.quadrature(0));
+  std::uniform_real_distribution<double> dist(-1.0, 1.0);
+  MAKE_GENERATOR(gen);
+  using flux_tags =
+      tmpl::list<Tags::Flux<Var1, tmpl::size_t<Dim>, Frame::Inertial>,
+                 Tags::Flux<Var2<Dim>, tmpl::size_t<Dim>, Frame::Inertial>>;
+  using div_tags = tmpl::list<
+      Tags::div<Tags::Flux<Var1, tmpl::size_t<Dim>, Frame::Inertial>>,
+      Tags::div<Tags::Flux<Var2<Dim>, tmpl::size_t<Dim>, Frame::Inertial>>>;
+
+  const auto logical_coords = logical_coordinates(mesh);
+  tnsr::I<DataVector, Dim, Frame::Inertial> inertial_coords{
+      mesh.number_of_grid_points()};
+  Jacobian<DataVector, Dim, Frame::Logical, Frame::Inertial> jacobian{
+      mesh.number_of_grid_points(), 0.0};
+  InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>
+      inverse_jacobian{mesh.number_of_grid_points(), 0.0};
+  for (size_t i = 0; i < Dim; ++i) {
+    jacobian.get(i, i) = 2.0;
+    inertial_coords.get(i) = 2.0 * logical_coords.get(i);
+  }
+
+  InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>
+      det_jac_times_inverse_jacobian{};
+
+  dg::metric_identity_jacobian(make_not_null(&det_jac_times_inverse_jacobian),
+                               mesh, inertial_coords, jacobian);
+  const auto det_jacobian = determinant(jacobian);
+  for (size_t i = 0; i < Dim; ++i) {
+    for (size_t j = 0; j < Dim; ++j) {
+      inverse_jacobian.get(i, j) =
+          det_jac_times_inverse_jacobian.get(i, j) / get(det_jacobian);
+    }
+  }
+
+  // Generate constant fluxes that aren't all the same.
+  Variables<flux_tags> fluxes{mesh.number_of_grid_points()};
+  const auto compute_smooth_fluxes = [](const auto fluxes_ptr,
+                                        const auto& coords) {
+    auto& local_fluxes = *fluxes_ptr;
+    tmpl::for_each<flux_tags>([&local_fluxes, &coords](auto tag_v) {
+      using tag = tmpl::type_from<decltype(tag_v)>;
+      for (size_t tensor_index = 0;
+           tensor_index < get<tag>(local_fluxes).size(); ++tensor_index) {
+        get<tag>(local_fluxes)[tensor_index] =
+            std::sqrt(tensor_index + 1) * square(get<0>(coords));
+        for (size_t d = 1; d < Dim; ++d) {
+          get<tag>(local_fluxes)[tensor_index] +=
+              std::sqrt(tensor_index + 1) * coords.get(d);
+        }
+      }
+    });
+  };
+  if (mesh.quadrature(0) == Spectral::Quadrature::GaussLobatto) {
+    fill_with_random_values(make_not_null(&fluxes), make_not_null(&gen),
+                            make_not_null(&dist));
+  } else {
+    compute_smooth_fluxes(make_not_null(&fluxes), inertial_coords);
+  }
+
+  Variables<div_tags> divergence_result{mesh.number_of_grid_points()};
+
+  weak_divergence(make_not_null(&divergence_result), fluxes, mesh,
+                  det_jac_times_inverse_jacobian);
```

Comment:
[nitpick for legibility] move the blank line on 219 down to below the `weak_divergence` line, to better separate the computation of the weak div from the expected div.